build.sbt

```diff
@@ -103,101 +112,125 @@ lazy val root = Project("geotrellis", file(".")).
   settings(unidocProjectFilter in (ScalaUnidoc, unidoc) := inAnyProject -- inProjects(geowave))
 
-lazy val macros = Project("macros", file("macros")).
-  settings(commonSettings: _*)
+lazy val macros = project
+  .settings(commonSettings)
 
-lazy val vectortile = Project("vectortile", file("vectortile"))
+lazy val vectortile = project
   .dependsOn(vector)
-  .settings(commonSettings: _*)
-
-lazy val vector = Project("vector", file("vector")).
-  dependsOn(proj4, util).
-  settings(commonSettings: _*)
+  .settings(commonSettings)
 
-lazy val vectorTest = Project("vector-test", file("vector-test")).
-  dependsOn(vector, vectorTestkit).
-  settings(commonSettings: _*)
+lazy val vector = project
+  .dependsOn(proj4, util)
+  .settings(commonSettings)
 
-lazy val vectorTestkit = Project("vector-testkit", file("vector-testkit")).
-  dependsOn(raster, vector).
-  settings(commonSettings: _*)
+lazy val `vector-test` = project
+  .dependsOn(vector, `vector-testkit`)
+  .settings(commonSettings)
 
-lazy val proj4 = Project("proj4", file("proj4")).
-  settings(commonSettings: _*).
-  settings(javacOptions ++= Seq("-encoding", "UTF-8"))
+lazy val `vector-testkit` = project
+  .dependsOn(raster, vector)
+  .settings(commonSettings)
 
-lazy val raster = Project("raster", file("raster")).
-  dependsOn(util, macros, vector).
-  settings(commonSettings: _*)
+lazy val proj4 = project
+  .settings(commonSettings)
+  .settings(javacOptions ++= Seq("-encoding", "UTF-8"))
 
-lazy val rasterTest = Project("raster-test", file("raster-test")).
-  dependsOn(raster, rasterTestkit, vectorTestkit).
-  settings(commonSettings: _*)
+lazy val raster = project
+  .dependsOn(util, macros, vector)
+  .settings(commonSettings)
 
-lazy val rasterTestkit = Project("raster-testkit", file("raster-testkit")).
-  dependsOn(raster, vector).
-  settings(commonSettings: _*)
+lazy val `raster-test` = project
+  .dependsOn(raster, `raster-testkit`, `vector-testkit`)
+  .settings(commonSettings)
 
-lazy val slick = Project("slick", file("slick")).
-  dependsOn(vector).
-  settings(commonSettings: _*)
+lazy val `raster-testkit` = project
+  .dependsOn(raster, vector)
+  .settings(commonSettings)
 
-lazy val spark = Project("spark", file("spark")).
-  dependsOn(util, vectortile, raster, rasterTestkit % "provided;test->test").
-  settings(commonSettings: _*)
+lazy val slick = project
+  .dependsOn(vector)
+  .settings(commonSettings)
+
+lazy val spark = project
+  .dependsOn(util, vectortile, raster, `raster-testkit` % "test")
+  .settings(commonSettings)
+  .settings(
+    // This takes care of a pseudo-cyclic dependency between the `spark` test scope, `spark-testkit`,
+    // and `spark` main (compile) scope. sbt is happy with this. IntelliJ requires that `spark-testkit`
+    // be added to the `spark` module dependencies manually (via "Open Module Settings" context menu for "spark" module).
+    unmanagedClasspath in Test ++= (fullClasspath in (LocalProject("spark-testkit"), Compile)).value
```

Comment:
Cool if it works (!)
There is a question appears what settings should be in the root `build.sbt` and what in local projects `build.sbt` files.